.github/workflows/build.yml

```diff
@@ -132,10 +133,28 @@ jobs:
         with:
           vs-version: '[17.9,)'
 
+      # Get changed components
+      - name: Get changed components
+        run: |
+          git fetch origin main
+          git diff --name-only origin/main...${{ github.sha }} -- components/ > ${{ github.workspace }}/changed-files.txt
+          $otherChanges = git diff --name-only origin/main...${{ github.sha }} | Select-String -NotMatch '^components/'
+          if (-not $otherChanges) {
+            $files = Get-Content "${{ github.workspace }}/changed-files.txt"
+            $names = $files | ForEach-Object { ($_ -replace '^components/', '') -replace '/.*$', '' }
+            $uniqueNames = $names | Sort-Object -Unique
+            $quotedNames = $uniqueNames | ForEach-Object { "'$_'" }
+            $changedComponentsList = $quotedNames -join ','
+            echo "CHANGED_COMPONENTS_LIST=$changedComponentsList" >> $env:GITHUB_ENV
+          }
+          else {
+            echo "CHANGED_COMPONENTS_LIST='all'" >> $env:GITHUB_ENV
+          }
+  
       # Generate full solution with all projects (sample gallery heads, components, tests)
       - name: Generate solution with ${{ matrix.multitarget }} gallery, components and tests
         working-directory: ./
-        run: powershell -version 5.1 -command "./tooling/GenerateAllSolution.ps1 -MultiTargets ${{ matrix.multitarget }} ${{ env.ENABLE_DIAGNOSTICS == 'true' && ' -UseDiagnostics' || '' }} -WinUIMajorVersion ${{ matrix.winui }}" -ErrorAction Stop
+        run: powershell -version 5.1 -command "./tooling/GenerateAllSolution.ps1 -MultiTargets ${{ matrix.multitarget }} ${{ env.ENABLE_DIAGNOSTICS == 'true' && ' -UseDiagnostics' || '' }} -Components ${{ env.CHANGED_COMPONENTS_LIST }} -WinUIMajorVersion ${{ matrix.winui }}" -ErrorAction Stop
```

Comment:
@Arlodotexe how does the input to the components parameter here in our script work with dependencies? i.e. if the only change is in X but that depends on component Y (in labs I guess), then is slngen/msbuild smart enough to know that Y is built too? I guess that'd mean we'd end up with packages for X and Y if only X changed. But that's not really a common thing we do here (compared to the main repo).

And I think this is just a process we'd use here in labs, so we could push things when changed to NuGet.org vs. in the main toolkit where we can just snap a preview to NuGet anyway and have nightlies in the custom feed if folks still need those, eh?

Reply:
> if the only change is in X but that depends on component Y (in labs I guess), then is slngen/msbuild smart enough to know that Y is built too?

Yes, you can specify a single component that references other components (such as SettingsControls in mainline WCT) and the dependencies (Extensions, Converters, etc) will get picked up. 

> I guess that'd mean we'd end up with packages for X and Y if only X changed. 

No, the package build script `./toolkit/Build-Toolkit-Components.ps1` doesn't use slngen in any capacity. It will only build the components you supply in the list, including nuget package and binlog output.